### ISSUE TITLE
Test result seemly wrong

### ISSUE THREAD

**Issue by JennyGao00** [MARSLab-UMN/TiltedImageSurfaceNormal#3]
Hi,
Recently, I have test your code on nyuv2 dataset, and run my test code as
## NYUv2
python train_test_surface_normal.py \
                     --checkpoint_path './checkpoints/DFPN_TAL.ckpt' \
                     --operation 'evaluate' \
                     --test_dataset 'nyud' \
                     --net_architecture 'dfpn' \
                     --batch_size 1
But, I got the wrong test result like this, 
![image](https://user-images.githubusercontent.com/55170462/118102315-3a425e00-b40b-11eb-9645-147c82eecf47.png)

I am quite confused, could you please explain why this is the case?
Looking forward to your reply.

**Comment by tiendo-zz** [comment]
For visualization, you need to normalize the output. What you show here seems to be a raw output from the network. You can use `torch.nn.functional.normalize` before visualize it.

**Comment by kvuong2711** [comment]
To supplement what @tiendo said, here I provided a sample code snippet that can be used to visualize the output from the network (matched NYU convention in your attached example). 
 
```
import torch
import numpy as np
import skimage.io as sio

def saving_normal_tensor_to_file(normal_tensor, path):
    normal_tensor = torch.nn.functional.normalize(normal_tensor, dim=0)
    output_normal_img = np.uint8((1 - normal_tensor.permute(1, 2, 0).detach().cpu()) * 127.5)
    temp = output_normal_img.copy()
    output_normal_img[:, :, 0] = temp[:, :, 2]
    output_normal_img[:, :, 1] = temp[:, :, 1]
    output_normal_img[:, :, 2] = temp[:, :, 0]
    sio.imsave(path, output_normal_img)
```

**Comment by JennyGao00** [comment]
OK, thanks very much!
